doc/lua_api.txt

```diff
@@ -5404,84 +5454,79 @@ Used by `minetest.register_node`, `minetest.register_craftitem`, and
             },
             damage_groups = {groupname = damage},
         },
+
+        -- If nil and item is node, prediction is made automatically
+        -- If nil and item is not a node, no prediction is made
+        -- If "" and item is anything, no prediction is made
+        -- Otherwise should be name of node which the client immediately places
+        -- on ground when the player places the item. Server will always update
+        -- actual result to client in a short moment.
         node_placement_prediction = nil,
-        --[[
-        ^ If nil and item is node, prediction is made automatically
-        ^ If nil and item is not a node, no prediction is made
-        ^ If "" and item is anything, no prediction is made
-        ^ Otherwise should be name of node which the client immediately places
-          on ground when the player places the item. Server will always update
-          actual result to client in a short moment.
-        ]]
+
+        -- if "", no prediction is made
+        -- if "air", node is removed
+        -- Otherwise should be name of node which the client immediately places
+        -- upon digging. Server will always update actual result shortly.
         node_dig_prediction = "air",
-        --[[
-        ^ if "", no prediction is made
-        ^ if "air", node is removed
-        ^ Otherwise should be name of node which the client immediately places
-          upon digging. Server will always update actual result shortly.
-        ]]
+
         sound = {
-            breaks = "default_tool_break", -- tools only
-            place = --[[<SimpleSoundSpec>]],
+            breaks = "default_tool_break",  -- tools only
+            place = <SimpleSoundSpec>,
         },
 
+        -- Shall place item and return the leftover itemstack
+        -- The placer may be any ObjectRef or nil.
+        -- default: minetest.item_place
         on_place = func(itemstack, placer, pointed_thing),
-        --[[
-        ^ Shall place item and return the leftover itemstack
-        ^ The placer may be any ObjectRef or nil.
-        ^ default: minetest.item_place ]]
+
+        -- Same as on_place but called when pointing at nothing.
+        -- The user may be any ObjectRef or nil.
```

Comment:
Should be `placer`, which is used for `on_place`, since the wielded item is _placed_, whereas `user` is used for `on_secondary_use` (below) because the wielded item is _used_.

Reply:
Oops, my bad >.<